### ISSUE TITLE
N0000035.RDB : Incorrect Door Position

### ISSUE THREAD

**Issue by Cliffworms** [Interkarma/daggerfall-unity#2171]
**Describe the bug**
In N0000035.RDB, there is a door that is placed halfway in the floor. See attached save game (unmodded) in one of the dungeons using this RDB.

**To Reproduce**
--Save game : 
[SAVE18.zip](https://github.com/Interkarma/daggerfall-unity/files/6795809/SAVE18.zip)

**Expected behavior**
The door should fit the door frame.

**Screenshots**
![2021_07_10_14_56_36](https://user-images.githubusercontent.com/83725484/125174074-c9ac8900-e190-11eb-8a5c-e3da599da8c4.jpg)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version 0.11.4

**Additional context**
I have stumbled upon this issue by working on the model list for the World Data Editor. Attached to this issue is a fixed version of N0000035.RDB for this door that I have done using the World Data Editor. 

[N0000035.RDB.json.zip](https://github.com/Interkarma/daggerfall-unity/files/6795811/N0000035.RDB.json.zip)
![2021_07_10_14_57_25](https://user-images.githubusercontent.com/83725484/125174162-535c5680-e191-11eb-94bb-02758f83f2b4.jpg)

**Comment by Cliffworms** [comment]
To add : 

Door model : 9005
Coordinates before : 
```
                        "XPos": 760,
                        "YPos": -256,
                        "ZPos": 360,
```
Coordinates after : 
```
                        "XPos": 760,
                        "YPos": -300,
                        "ZPos": 360,
```

**Comment by ajrb** [comment]
We need to decide whether these kinds of dungeon block errors should be fixed by the runtime data patching code that Ferital has done, or whether it should be part of fix mods such as XJDHDR's unofficial block fixes mod. The advantage of putting it in the DFU code is that the whole block doesn't need a JSON world data override to fix one coordinate. 

@df-ferital what do you think?

**Comment by KABoissonneault** [comment]
IMO we should do code so players can have the "out of the box" working experience. 

In addition, I think we should have a setting to turn off *all* engine-side block modifications, including the existing ones. The setting doesn't have to be in the UI, having it INI only would be fine. Just as a backdoor in case someone wants to make a mod to fix it themselves

**Comment by ajrb** [comment]
> In addition, I think we should have a setting to turn off _all_ engine-side block modifications, including the existing ones. The setting doesn't have to be in the UI, having it INI only would be fine. Just as a backdoor in case someone wants to make a mod to fix it themselves

I disagree, having runtime fixes of clearly erroneous values does not stop a mod from modifying them. There's no need to turn off these corrections for any reason that I can think of.

**Comment by df-ferital** [comment]
@ajrb I agree that it would be better to have all these fixes as a runtime patch. I could correct this one but there are many errors like this throughout BSA files. That's why for now, I mostly focused on texture issues and game breaking data such as invalid dungeon blocks. Maybe @XJDHDR could put this fix in his unofficial patch for now, until I have some time to put it back in my runtime hacks.

**Comment by ajrb** [comment]
@df-ferital I agree this is a good candidate, also with your priorities. Would you mind if I added fixes like this, or would you prefer to keep full control? This type of fix I understand, wheras the texture issues and UV etc I am clueless about and very glad you've been doing such a great job of correcting them.

**Comment by df-ferital** [comment]
@ajrb Of course, don't hesitate to push a fix for this, no problem on my side.

**Comment by ajrb** [comment]
Fixed by PR #2175
